### PR TITLE
Fix cvc5-projects issue 497

### PR DESCRIPTION
**Pull request by mudathirmahgoub**
Fixes https://github.com/cvc5/cvc5-projects/issues/497

### CODE REVIEW

**Comment by ajreynol** [comment]
I don't understand why this fix is necessary, it seems that there is a more critical issue that is being hidden here.  Is there a way to fix this at its source?

File: src/theory/bags/card_solver.cpp [hunk 1]
```diff
@@ -331,12 +331,21 @@ void CardSolver::checkLeafBag(const std::pair<Node, Node>& pair,
...
       }
     }
   }
 }
 
+void CardSolver::sendInferInfo(InferInfo& inferInfo)
+{
+  if (!inferInfo.isCachedLemma())
+  {
+    d_hasNewPendingLemmas = true;
```
Comment (ajreynol) [inline src/theory/bags/card_solver.cpp:344]: I don't think this is accurate.  You are actually sending a lemma here, not adding it to pending.

File: src/theory/bags/card_solver.cpp [hunk 2]
```diff
@@ -102,31 +98,33 @@ void CardSolver::checkCardinalityGraph()
         case BAG_DIFFERENCE_REMOVE: checkDifferenceRemove(pair, n); break;
         default: break;
       }
+      if (d_hasNewPendingLemmas)
```
Comment (ajreynol) [inline src/theory/bags/card_solver.cpp:101]:
```suggestion
      if (d_im.hasSentLemma())
```

**Review by ajreynol — CHANGES_REQUESTED** [review]
You shouldn't have to track whether you have a pending lemma, since the inference managers do this for you.  I think the code can be simplified accordingly, based on the change to check `d_im.hasSentLemma()`.

File: src/theory/bags/card_solver.cpp (hunk 1 above)
Comment (mudathirmahgoub) [reply src/theory/bags/card_solver.cpp]: Done. Actually I hated the new boolean.

File: src/theory/bags/card_solver.cpp (hunk 2 above)
Comment (mudathirmahgoub) [reply src/theory/bags/card_solver.cpp]: Done

File: src/theory/bags/infer_info.h [hunk 3]
```diff
@@ -72,6 +74,14 @@ class InferInfo : public TheoryInference
...
    */
   bool isFact() const;
+  /**
+   * @return the lemma for this InferInfo.
+   */
+  Node getLemma() const;
+  /**
+   * @return whether this InferInfo represents a cached lemma.
+   */
+  bool isCachedLemma() const;
```
Comment (ajreynol) [inline src/theory/bags/infer_info.h:84]: Can probably just delete this too right?

**Review by ajreynol — APPROVED** [review]
One minor comment.

**Comment by ajreynol** [comment]
Can you rename the regression `regress1/bags/murxla7.smt2` to match the project issue #?

File: src/theory/bags/infer_info.h (hunk 3 above)
Comment (mudathirmahgoub) [reply src/theory/bags/infer_info.h]: Done.

**Comment by mudathirmahgoub** [comment]
> Can you rename the regression `regress1/bags/murxla7.smt2` to match the project issue #?

Done

**Pull request merged**